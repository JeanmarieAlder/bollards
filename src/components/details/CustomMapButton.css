.map-expand-button {
    height: 60px;
    width: 120px;
    font-size: 24px;
}