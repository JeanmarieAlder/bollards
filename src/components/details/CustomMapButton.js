import { Button } from 'react-bootstrap';
import React, { useState } from 'react';
import { useMap, useMapEvents } from 'react-leaflet';
import { useTranslation } from "react-i18next";
import { Link } from 'react-router-dom';

import './CustomMapButton.css';

const CustomMapButton = () => {
    const { t } = useTranslation();
    const map = useMap();
    const [zoomLevel, setZoomLevel] = useState(map.getZoom());
    const [latLng, setLatLng] = useState(map.getCenter());

    const mapEvents = useMapEvents({
        zoomend: () => {
            setZoomLevel(mapEvents.getZoom());
        },
        moveend: () => {
            setLatLng(mapEvents.getCenter());
        }
    })
    return (
        <div className={"topright leaflet-top leaflet-right"} style={{height: "80px", width: "200px"}} >
            <div className={"leaflet-control"}>
                <Link to={`/map?zoom=${zoomLevel}&lat=${latLng.lat}&lon=${latLng.lng}`}>
                <Button 
                    id="expand-map-button" 
                    variant="dark" 
                    className="map-expand-button"
                    ><b>{t("s_expand")}</b></Button>
            </Link>
            </div>
        </div>
        
    )
};

export default CustomMapButton;