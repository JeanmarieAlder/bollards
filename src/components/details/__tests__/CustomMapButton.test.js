import { render } from '@testing-library/react';
import React from 'react';
import { MapContainer } from 'react-leaflet';
import { BrowserRouter } from 'react-router-dom';
import CustomMapButton from '../CustomMapButton';

jest.mock('react-i18next', () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
      return {
        t: (str) => str,
        i18n: {
          ...jest.requireActual('react-i18next'),
          on: () => {}
        },
      };
    },
  }));

function mockFunction() {
    const original = jest.requireActual('react-router-dom');
    return {
        ...original,
        useLocation: () => ({
        pathname: '/another-route',
        search: '',
        hash: '',
        state: null,
        key: '5nvxpbdafa',
        })
    };
} 

jest.mock('react-router-dom', () => mockFunction());

test("Component displays correctly", () => {
    const component = render(
        <BrowserRouter>
            <MapContainer 
                style={{height:'100%'}}
                center={[46.12, 7.12]} 
                zoom={15}>
                <CustomMapButton></CustomMapButton>
            </MapContainer>
        </BrowserRouter>
        
      );
      let tree = component.asFragment();
      expect(tree).toMatchSnapshot();
});