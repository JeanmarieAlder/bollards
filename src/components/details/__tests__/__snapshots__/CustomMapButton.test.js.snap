// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component displays correctly 1`] = `
<DocumentFragment>
  <div
    class="leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
    style="height: 100%; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      />
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-disabled="false"
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            <span
              aria-hidden="true"
            >
              +
            </span>
          </a>
          <a
            aria-disabled="false"
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            <span
              aria-hidden="true"
            >
              âˆ’
            </span>
          </a>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JavaScript library for interactive maps"
          >
            Leaflet
          </a>
        </div>
      </div>
    </div>
    <div
      class="topright leaflet-top leaflet-right"
      style="height: 80px; width: 200px;"
    >
      <div
        class="leaflet-control"
      >
        <a
          href="/map?zoom=15&lat=46.12&lng=7.12"
        >
          <button
            class="map-expand-button btn btn-dark"
            id="expand-map-button"
            type="button"
          >
            <b>
              s_expand
            </b>
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
